### PR TITLE
Add payment type: Sofort

### PR DESCRIPTION
**Pull request by lsl**
#### Changes proposed in this Pull Request

* Add the Sofort payment type per D21171-code 
* Add the Sofort logo & styles
* Add support for `ORDER_TRANSACTION_STATUS.ASYNC_PENDING`
* Configure redirect payment thank you page to route to pending payments on an async pending order status
* Add a banner to pending payments page with further payment info

#### Testing instructions
* Same as testing other redirect payment types but there is a secondary event you must receive to finalize the pending payment.
* Success redirection - use `succeeding_charge` as the name to make the transaction migrate from `pending` to `succeeded` (+notification)
* Failure redirection - use `failing_charge` as the name to make the transaction migrate from `pending` to `failed ` (+notification)

#### Checkout flow (as seen on calypso)
1. Capture name and move on to Sofort.
sofort-payment-type
2. Return to calypso with a pending redirect
pending-payments-processing-before
3. Finally end up on the pending purchases screen.
pending-payments-after

### CODE REVIEW

**Comment by yoavf** [comment]
Not sure we need the sofort payment box. We can probably use the regular redirect payment box.

**Comment by yoavf** [comment]
Let's make sure we're redirected to the right place where the payment is done, and display the right message to the customers.

Right now we go to the processing payment page, but will be redirected back to plans as soon as the status on the backend changes to `async-pending`.

File: client/my-sites/checkout/checkout/secure-payment-form.jsx [hunk 1]
```diff
@@ -319,6 +319,29 @@ const SecurePaymentForm = createReactClass( {
 		);
 	},
 
+	renderSofortPaymentBox() {
```
Comment (yoavf) [inline client/my-sites/checkout/checkout/secure-payment-form.jsx]: leftover, no longer needed

File: client/my-sites/checkout/checkout-thank-you/pending.jsx [hunk 2]
```diff
@@ -97,7 +97,22 @@ class CheckoutPending extends PureComponent {
...
-		const { orderId, siteSlug, translate } = this.props;
+		const { orderId, siteSlug, translate, transaction } = this.props;
+
+		if ( transaction && ORDER_TRANSACTION_STATUS.ASYNC_PENDING === transaction.processingStatus ) {
+			return (
+				<Main className="checkout-thank-you__pending">
+					<EmptyContent
+						illustration={ '/calypso/images/illustrations/illustration-shopping-bags.svg' }
+						illustrationWidth={ 500 }
+						title={ translate( 'Payment pending…' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/pending.jsx]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Payment pending' )` *ES Score: 20*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/payment-type-sofort)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/me/pending-payments/index.jsx [hunk 3]
```diff
@@ -83,11 +84,22 @@ export class PendingPayments extends Component {
...
-				<div>
-					{ pendingPayments.map( purchase => (
-						<PendingListItem key={ purchase.siteId } { ...purchase } />
-					) ) }
-				</div>
+				<React.Fragment>
+					<Banner
+						callToAction={ translate( 'Back to My Sites' ) }
+						description={ translate(
+							'Your payment initiation has been confirmed. We are currently waiting for the funds to clear.'
```
Comment (yoavf) [inline client/me/pending-payments/index.jsx]: Let's add a timeline here too (see [my comment here](https://github.com/Automattic/wp-calypso/pull/30260#discussion_r249417176))

**Pull request merged**